src/components/VideoPlayerContexts/VideoPopoverMenuContext.js

```diff
@@ -0,0 +1,78 @@
+import PropTypes from 'prop-types';
+import React, {useCallback, useContext, useMemo, useState} from 'react';
+import _ from 'underscore';
+import * as Expensicons from '@components/Icon/Expensicons';
+import useLocalize from '@hooks/useLocalize';
+import fileDownload from '@libs/fileDownload';
+import * as Url from '@libs/Url';
+import CONST from '@src/CONST';
+import {usePlaybackContext} from './PlaybackContext';
+
+const VideoPopoverMenuContext = React.createContext(null);
+
+function VideoPopoverMenuContextProvider({children}) {
+    const {currentVideoPlayerRef} = usePlaybackContext();
+    const {translate} = useLocalize();
+    const [currentPlaybackSpeed, setCurrentPlaybackSpeed] = useState(CONST.VIDEO_PLAYER.PLAYBACK_SPEEDS[CONST.VIDEO_PLAYER.DEFAULT_PLAYBACK_SPEED_INDEX]);
+
+    const updatePlaybackSpeed = useCallback(
+        (speed) => {
+            setCurrentPlaybackSpeed(speed);
+            currentVideoPlayerRef.current.setStatusAsync({rate: speed});
+        },
+        [currentVideoPlayerRef],
+    );
+
+    const downloadAttachment = useCallback(() => {
+        currentVideoPlayerRef.current.getStatusAsync().then((status) => {
+            const sourceURI = `/${Url.getPathFromURL(status.uri)}`;
+            fileDownload(sourceURI);
+        });
+    }, [currentVideoPlayerRef]);
+
+    const menuItems = useMemo(
+        () => [
+            {
+                icon: Expensicons.Download,
+                text: translate('common.download'),
+                onSelected: () => {
+                    downloadAttachment();
+                },
+            },
+            {
+                icon: Expensicons.Meter,
+                text: translate('videoPlayer.playbackSpeed'),
+                subMenuItems: [
+                    ..._.map(CONST.VIDEO_PLAYER.PLAYBACK_SPEEDS, (speed) => ({
+                        icon: currentPlaybackSpeed === speed ? Expensicons.Checkmark : null,
+                        text: speed.toString(),
+                        onSelected: () => {
+                            updatePlaybackSpeed(speed);
+                        },
+                        shouldPutLeftPaddingWhenNoIcon: true,
```

Comment:
Not sure if this is the culprit, but is there a way to make sure all options are aligned, or put the checkmark on the right instead?


<img width="351" alt="image" src="https://github.com/Expensify/App/assets/2229301/e5f5dd9d-490b-423f-a647-812423ed7155">

Reply:
The margin is being explicitly added here https://github.com/Expensify/App/blob/main/src/components/MenuItem.tsx#L318

Not sure if we are good with this, or should I add an prop that disables it in our case.

Reply:
Yeah so the original mockups we made in the design doc had the checkmark on the right.

<img width="425" alt="image" src="https://github.com/Expensify/App/assets/2229301/0f822310-abfc-4aa3-8fc2-bdc06d496ba7">

I can't tell if there's a proper way to move the icon to the right with `MenuItem`s, except maybe with the props `shouldShowRightIcon`/`iconRight` or `shouldShowRightComponent`/`rightComponent`, but it sounds slightly hacky.

If you can think of a way to fix the margin with the checkmark icon on the left then I would say that's fine, but cc @dannymcclain in case you have a strong opinion to put it on the right.

Reply:
Ok no strong preference - let's go with whatever's easier to do but let's remove this gap on the left of the selected value then?

<img width="359" alt="image" src="https://github.com/Expensify/App/assets/2229301/e389459d-1409-4d84-8cf8-317a6b313eeb">